cli/src/cmd/cast/call.rs

```diff
@@ -0,0 +1,67 @@
+use crate::opts::{
+    cast::{parse_block_id, parse_name_or_address},
+    EthereumOpts,
+};
+
+use clap::Parser;
+use ethers::types::{BlockId, NameOrAddress};
+use eyre::Result;
+use foundry_config::{
+    figment::{
+        self,
+        value::{Dict, Map, Value},
+        Metadata, Profile,
+    },
+    find_project_root_path, Config,
+};
+
+use serde::Serialize;
+
+#[derive(Debug, Clone, Parser, Serialize)]
+pub struct CallArgs {
+    #[clap(help = "the address you want to query", parse(try_from_str = parse_name_or_address))]
+    #[serde(skip)]
+    pub address: NameOrAddress,
+    #[serde(skip)]
+    pub sig: String,
+    #[serde(skip)]
+    pub args: Vec<String>,
+    #[clap(long, short, help = "the block you want to query, can also be earliest/latest/pending", parse(try_from_str = parse_block_id))]
+    #[serde(skip)]
+    pub block: Option<BlockId>,
+    #[clap(flatten)]
+    #[serde(flatten)]
+    pub eth: EthereumOpts,
+}
+
+impl<'a> From<&'a CallArgs> for Config {
+    fn from(args: &'a CallArgs) -> Self {
+        let config = Config::figment_with_root(find_project_root_path().unwrap())
+            .merge(args)
+            .extract::<Config>()
+            .unwrap_or_else(|err| panic!("{}", err));
+
+        config
+    }
+}
```

Comment:
have a look at the forge build command which does this in two steps.
but I suppose you could also use the `impl_figment_convert!(CallArgs);` here?

Reply:
Yes, so `impl_figment_convert` requires CallArgs to have a `root` field, but that doesn't seem appropriate for call right? What can be done here?

Reply:
I see, then lets mirror what the `impl_figment_convert` macro does adapted for without root dir?